gobblin-test-harness/src/main/java/gobblin/test/execution/validator/ValidationRule.java

```diff
@@ -0,0 +1,31 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.test.execution.validator;
+
+
+
+/**
+ *  An interface is for defining the validator for the test. The input can be the set of input files and the output will be be output of 
+ * 
+ * @author sveerama
+ *
+ * @param <I> input data files or config
+ * @param <O> Output data files
+ */
+
+public interface ValidationRule<I,O> {
```

Comment:
I think we can lose the generic types "I" and "O". `isValid` doesn't need to take in parameters. Any required information for a `ValidationRule` can just be passed in via the constructor.